version: "3.8"

services:
  client:
    build: .
    image: eligibility_verification_client:latest
    entrypoint: ["bin/entrypoint.sh"]
    command: manage runserver 0.0.0.0:8000
    environment:
      - DJANGO_SECRET_KEY
      - DJANGO_DEBUG
      - DJANGO_TRANSIT_AGENCY
      - DJANGO_DB
      - POSTGRES_DB
      - POSTGRES_USER
      - POSTGRES_PASSWORD
      - POSTGRES_HOST
      - POSTGRES_PORT
    ports:
      - "${DJANGO_LOCAL_PORT}:8000"
    volumes:
      - .:/usr/src/cal-itp:cached

  db:
    image: postgres:12
    environment:
      - POSTGRES_DB
      - POSTGRES_USER
      - POSTGRES_PASSWORD

  pgadmin:
    image: dpage/pgadmin4
    environment:
      - PGADMIN_DEFAULT_EMAIL=example@example.com
      - PGADMIN_DEFAULT_PASSWORD=pgadmin
    ports:
      - "8001:80"

  server:
    build: ./server
    image: eligibility_verification_server:dev
    command: python app.py
    environment:
      - FLASK_APP=app.py
      - FLASK_DEBUG=1
      - FLASK_ENV=development
    ports:
      - "5000:5000"
    volumes:
      - ./server:/usr/src/server:cached
      - ./data/server:/usr/src/server/data